javaslang/src/test/java/javaslang/collection/euler/Euler41Test.java

```diff
@@ -0,0 +1,52 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection.euler;
+
+import javaslang.collection.List;
+import javaslang.collection.Stream;
+import static javaslang.collection.euler.Utils.isPrime;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class Euler41Test {
+
+    /**
+     * <strong>Problem 41 Pandigital prime</strong>
+     * <p>
+     * We shall say that an <i>n</i>-digit number is pandigital if it makes use
+     * of all the digits 1 to <i>n</i> exactly once. For example, 2143 is a
+     * 4-digit pandigital and is also prime.
+     *
+     * What is the largest <i>n</i>-digit pandigital prime that exists?
+     * <p>
+     * See also <a href="https://projecteuler.net/problem=41">projecteuler.net
+     * problem 41</a>.
+     */
+    @Test
+    public void shouldSolveProblem41() {
+        assertThat(nDigitPandigitalNumbers(4)).contains(2143);
+        assertThat(isPrime(2143)).isTrue();
+
+        assertThat(largestNPandigitalPrime()).isEqualTo(7652413);
+    }
+
+    private static int largestNPandigitalPrime() {
+        return Stream.rangeClosedBy(9, 1, -1)
+                .flatMap(n -> nDigitPandigitalNumbers(n)
+                        .filter(Utils::isPrime)
+                        .sorted()
+                        .reverse())
```

Comment:
Yep, works fine and looks good. Thanks

On 18 August 2016 at 21:07, Daniel Dietrich <notifications@github.com>
wrote:

> In javaslang/src/test/java/javaslang/collection/euler/Euler41Test.java
> <https://github.com/javaslang/javaslang/pull/1508#discussion_r75369799>:
>
> > +     * problem 41</a>.
> > +     */
> > +    @Test
> > +    public void shouldSolveProblem41() {
> > +        assertThat(nDigitPandigitalNumbers(4)).contains(2143);
> > +        assertThat(isPrime(2143)).isTrue();
> > +
> > +        assertThat(largestNPandigitalPrime()).isEqualTo(7652413);
> > +    }
> > +
> > +    private static int largestNPandigitalPrime() {
> > +        return Stream.rangeClosedBy(9, 1, -1)
> > +                .flatMap(n -> nDigitPandigitalNumbers(n)
> > +                        .filter(Utils::isPrime)
> > +                        .sorted()
> > +                        .reverse())
>
> we could omit the reverse by providing a comparator:
>
> sorted(Comparator.reverseOrder())
>
> (does that compile?)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/javaslang/javaslang/pull/1508/files/3a432624d3a68fcfe04c5726f2bd2e65e5e4b616#r75369799>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADRVwDZa-xZUycsGA5kM97ycn8u8Eb2Vks5qhK1vgaJpZM4JnwXL>
> .
>